src/algo/collection_ext/mod.rs

```diff
@@ -160,15 +166,14 @@ pub trait CollectionExt: Collection {
     where
         Pred: Fn(&Self::Element) -> bool,
     {
-        let mut start = self.start();
-        let end = self.end();
-        while start != end {
-            if pred(self.at(&start)) {
-                return start;
+        let mut rest = self.all();
+        while let Some(x) = rest.first() {
+            if pred(x) {
+                break;
             }
-            start = self.after(start);
+            rest.pop_first();
```

Comment:
It would be more efficient to use a `drop_first` if the cost of returning the first element is nontrivial.